### PR TITLE
ENT-3970: Fix sys.flavor on Aix to not include minor version

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a PR! Maybe @olehermanse can review this?

**Comment by nickanderson** [comment]
@cf-bottom jenkins, with exotics, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1657/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1657/

File: libenv/sysinfo.c [hunk 1]
```diff
@@ -1352,7 +1352,14 @@ static void OSClasses(EvalContext *ctx)
 #else
 
     char vbuff[CF_MAXVARSIZE];
+
+#ifdef _AIX
+    strlcpy(vbuff, VSYSNAME.version, CF_MAXVARSIZE);
+#endif
+#ifndef _AIX
```
Comment (vpodzime) [inline libenv/sysinfo.c]: You can do `#else` here or if that doesn't work use `#if defined(_AIX)...#else...#endif`.

**Comment by nickanderson** [comment]
@cf-bottom jenkins me, with exotics

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1684/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1684/

**Comment by vpodzime** [comment]
Failed on AIX -- maybe `\d` is too fancy for the regexp? Or does it need to be double-escaped, maybe?

File: libenv/sysinfo.c (hunk 1 above)
Comment (nickanderson) [reply libenv/sysinfo.c]: DONE

**Comment by nickanderson** [comment]
@cf-bottom jenkins, with exotics

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1686/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1686/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1758/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1758/

**Comment by nickanderson** [comment]
:panda_face: 
11:40:01 ./01_vars/01_basic/sys_flavor_on_aix_no_dots.cf FAIL (UNEXPECTED FAILURE)

**Comment by nickanderson** [comment]
Ah, found it. The failures are because of a bug in the dcs_check_regcmp bundle from the test framework.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, with exotics, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1763/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1763/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, with exotics, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1764/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1764/

**Comment by nickanderson** [comment]
Finally `01:06:19.921 ./01_vars/01_basic/sys_flavor_on_aix_no_dots.cf Pass` on AIX.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, with exotics, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1765/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1765/

**Pull request merged**